src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -242,9 +246,13 @@ public override byte[] ToBinary(object obj)
         /// <returns>The object contained in the array</returns>
         public override object FromBinary(byte[] bytes, Type type)
         {
-            string data = Encoding.UTF8.GetString(bytes);
-            object res = JsonConvert.DeserializeObject(data, Settings);
-            return TranslateSurrogate(res, this, type);
+            using (var stream = new MemoryStream(bytes))
+            using (var r = new StreamReader(stream))
+            using (var reader = new JsonTextReader(r))
+            {
+                object res = _serializer.Deserialize(reader);
```

Comment:
So the `Settings` are passed in properly here when the `_serializer` is created - is the issue that the `_serializer` class isn't meant to be used concurrently or is it that `JsonConvert` is doing something differently than `_serializer.Deserialize`?

Reply:
Looking into this now

Reply:
Thanks!

Reply:
Ah, looks like the `StreamWriter` used in `ToBinary` above needs to be flushed/disposed before attempting to return the `byte[]`. Will push an updated commit after re-running tests locally.

Reply:
Can you re-run the benchmarks also?